### ISSUE TITLE
Add option to collect GHC eventlog

### ISSUE THREAD

**Issue by TeofilC** [haskell/cabal#8621]
#### Summary 

GHC supports outputting information to the eventlog (see: https://www.haskell.org/ghc/blog/20190924-eventful-ghc.html). Normally collecting the eventlog from a haskell application is as simple as passing `+RTS -lout.eventlo -RTS` and then you get an eventlog at `out.eventlog`. So, you might expect that adding that to `ghc-options` would allow you to collect an eventlog from Cabal's invocation of GHC. But this doesn't work because Cabal invokes GHC multiple times, so, subsequent invocations will overwrite previous ones.

#### Suggestion
Add a new option that will pass flags to GHC to write eventlogs to a well-known location in the work dir. It would ensure that eventlogs don't get overwritten by ensuring that the paths they are writing to are made unique.
The option could be called `ghc-collect-eventlog`.

#### Workarounds
If people want eventlog's from GHC in practice I've mostly seen them write wrapper scripts that invoke GHC in a way to ensure eventlogs don't get overwritten. EG, head.hackage calls it in a way that the `-l` option is only passed for the `--make` call and includes which unit is being compiled. https://gitlab.haskell.org/ghc/head.hackage/-/blob/master/ci/logging-ghc

**Comment by ulysses4ever** [comment]
This sounds plausible but let me clarify: who is the target audience of this feature? Is it mostly needed for GHC developers to figure out compiler's resource consumption and the like? In that case, it sounds like a pretty narrow user base. I'm not saying it's not worthy, just that maybe it's not high priority. We'd certainly welcome a PR is someone creates it, I think.

**Comment by TeofilC** [comment]
Good point @ulysses4ever . I think you are right that GHC devs are the primary people that are concerned about this. But I think as well there's a secondary audience in people who are trying to investigate the cause of slow builds. I agree it's not high priority.

I'll try to find some time to make a PR to fix this at some point
